packaging/fedora/uberftp.spec

```diff
@@ -8,7 +8,8 @@ Summary:        GridFTP-enabled ftp client
 
 License:        NCSA
 URL:            https://github.com/gridcf/UberFTP
-Source0:        https://github.com/gridcf/UberFTP/archive/%{gittag}/%{name}-%{version}.tar.gz
+#Source0:        https://github.com/gridcf/UberFTP/archive/%{gittag}/%{name}-%{version}.tar.gz
+Source0:        https://github.com/gridcf/UberFTP/archive/%{gittag}/uberftp-client-2.9-Beta.tar.gz
```

Comment:
I'm a bit confused about this one?
Firstly where is the `-client` coming from (after all there is no server)? And is it wise to hardcode the name and version there?
Also it doesn't currently exist, is it produced as part of a still running build? (Sorry for my ignorance)

Reply:
The original file name (and URL) was meant to build from the source tarball as provided by GitHub (i.e. the untampered contents of the Git repo without the `.git` dir). For the RPM builds on Travis-CI the actual dist tarball is created prior and used instead. And due to the current configuration in `configure.ac`:
```
[...]
AC_INIT(UberFTP (2.9 Beta), [2.9-Beta], [Grid Community Forum <discuss@gridcf.org>],[uberftp-client])
[...]
```
...`make dist` creates a tarball named `uberftp-client-2.9-Beta.tar.gz`. The URL in the RPM spec is ignored by `rpm-build` as long as the source file name matches IIUIC.

This needs some rethinking about what we want to have in the RPM spec file. I think, as soon as we upload the dist tarballs to `repo.gridcf.org` (would be my plan for the next PR, but needs help from OSG people with the encrypted credentials to be used by the GitHub action) we should just use a URL pointing there. For the naming we should include a command line in the RPM spec that determines the relevant name parts from the `configure.ac`. So on version changes we only need to change things there then.

Reply:
I think it is best to rename the `uberftp-client` in the `configure.ac` into uberftp as that's how we have it in EPEL and I think dist tarballs have never been released.
Then I would put the name, the version and release manually one time in the spec file (something like [here](https://src.fedoraproject.org/rpms/curl/blob/rawhide/f/curl.spec#_2-4) and derive all other occurrences in the spec file from there. You could use magic to get it from the configure.ac but I'm not sure that's worth the effort.

Reply:
Sounds good! I'll take this into account and push it in an additional change.

Reply:
About the version number. I agree, that it's best to hardcode it in the spec file, as deriving it from the `configure.ac` might not even work during runtime assuming that the sources aren't yet extracted when `rpm-build` starts processing the spec file. Late yesterday I thought about a `VERSION` file in the UberFTP sources and use `m4` to get it into the configure.ac during runtime (e.g. [this description](https://newbedev.com/read-a-version-number-from-a-file-in-configure-ac) looks like a good option, especially the variant which auto-includes  the `VERSION` file into the `Makefile`). Then we would need to change it "only" in the `VERSION` file and the RPM spec for releases.

Reply:
In which case I would definitely go for the `m4_normalize(m4_include([VERSION]))` way of doing it.

Reply:
Addressed with https://github.com/gridcf/UberFTP/pull/16/commits/59bdc2b0cd195516a7a8187e295ae9ce555f19fe